src/features/import/hooks/useGenderMapping.ts

```diff
@@ -0,0 +1,84 @@
+import { columnUpdate } from '../store';
+import { useAppDispatch } from 'core/hooks';
+import { CellData, Column, ColumnKind } from '../utils/types';
+
+export const genders = ['f', 'm', 'o'] as const;
+export type Gender = typeof genders[keyof typeof genders];
+
+export default function useGenderMapping(column: Column, columnIndex: number) {
+  const dispatch = useAppDispatch();
+
+  const getSelectedGender = (value: CellData) => {
+    if (column.kind == ColumnKind.GENDER) {
+      const map = column.mapping.find((m) => m.value === value);
+      return map?.gender ?? null;
+    }
+    return null;
+  };
+
+  const selectGender = (gender: Gender, value: CellData) => {
+    if (column.kind !== ColumnKind.GENDER) {
+      return;
+    }
+
+    // Check if there is already a map for this row value to an org ID
+    const map = column.mapping.find((map) => map.value == value);
+    // If no map for that value
+    if (!map) {
+      const newMap = { gender, value: value };
+      dispatch(
+        // Add value to mapping for the column
+        columnUpdate([
+          columnIndex,
+          {
+            ...column,
+            mapping: [...column.mapping, newMap],
+          },
+        ])
+      );
+    } else {
+      // If there is already a map, replace it
+      // Find mappings that are not for this row value
+      const filteredMapping = column.mapping.filter((m) => m.value != value);
+      // New orgId for that row value
+      const updatedMap = { ...map, gender };
```

Comment:
Some little copy-paste mistakes here in the comments, do you wanna updated them to relate to gender mapping and not org mapping? :)

Reply:
I think this is a perfect example of the reasons why I try to avoid comments. They exist to make the code easier to read, but it's so easy to make mistakes like this where the code actually becomes more confusing because of the comment.

Generally, if I feel like a comment is needed, I will try to rewrite the code (and especially _name things_) so that the code itself becomes easier to read and the comment isn't needed.

Reply:
I think the code was actually way too complicated for what it was doing. I changed the implementation. Let me know what you think!